python/social_bot/envs/gazebo_base.py

```diff
@@ -133,6 +133,48 @@ def render(self, mode='human'):
     def close(self):
         super().close()
         gazebo.close_without_model_base_fini()
+    
+    def insert_model(self, model, name=None, pose="0 0 0 0 0 0"):
+        """
+        Insert a model with into a specific position of the world
+        Args:
+            model (string): the name of the model in the model database
+            name (string): the name of the model in the world
+            pose (string): the pose of the model, format is "x y z roll pitch yaw"
+        """
+        if name == None:
+            name = model
+        model_sdf = """
+        <?xml version='1.0'?>
+        <sdf version ='1.4'>
+        <model name=""" + name + """>
+            <include>
+                <uri>model://""" + model + """</uri>
+            </include>
+            <pose frame=''>""" + pose + """</pose>
+        </model>
+        </sdf>
+        """
+        self._world.insertModelFromSdfString(model_sdf)
+        # Sleep for a while waiting for gzserver to finish the inserting
+        # operation. Or it may not be successfully inserted and report error.
+        time.sleep(0.2)
+        self._world.step(20)
+
+    def insert_model_list(self, model_list):
+        """
+        Insert models into the world
+        Args:
+            model_list (list) : the list of the models
+        """
+        obj_num = len(model_list)
+        for obj_id in range(obj_num):
+            model_name = model_list[obj_id]
+            if self._world.model_list_info().find(model_name) == -1:
+                self._world.insertModelFile('model://' + model_name)
+                logging.debug('model ' + model_name + ' inserted')
+                time.sleep(0.2)
+                self._world.step(20)
```

Comment:
This is sleeping and stepping for every model inserted into the world?  Probably only need to do that after all models are inserted?  Only potential problem would be if the model list contains duplicates, then the same item will be inserted twice?  But that's probably a problem of the input list?

Reply:
yes, it's a must. Each insertion need to wait for some time or gazebo crashes....
There is a checking for duplicates, line 173 
if self._world.model_list_info().find(model_name) == -1

Reply:
but this still is buggy. Like, if we instead an appletree first, we can not insert a tree latter..